src/openvic-simulation/map/MapDefinition.cpp

```diff
@@ -637,6 +659,196 @@ bool MapDefinition::load_map_images(fs::path const& province_path, fs::path cons
+				if (river_data[idx + 1] == size + 1) {
+					points.push_back({ points.back().x + 1, points.back().y });
+					direction = 3;
+					continue;
+				} else if (river_data[idx + 1] == MERGE_COLOUR) {
+					merge_location = { points.back().x + 1, points.back().y };
+					merge = true;
+				} else if (river_data[idx + 1] > 1 && river_data[idx + 1] < 12) { // new segment
+					new_point = { points.back().x + 1, points.back().y };
+					direction = 3;
+					break;
+				}
+			}
+			// check pixel below
+			if (points.back().y < rivers_bmp.get_height() - 1 && direction != 2) {
+				if (river_data[idx + rivers_bmp.get_width()] == size + 1) {
+					points.push_back({ points.back().x, points.back().y + 1 });
+					direction = 1;
+					continue;
+				} else if (river_data[idx + rivers_bmp.get_width()] == MERGE_COLOUR) {
+					merge_location = { points.back().x, points.back().y + 1 };
+					merge = true;
+				} else if (river_data[idx + rivers_bmp.get_width()] > 1 && river_data[idx + rivers_bmp.get_width()] < 12) { // new segment
+					new_point = { points.back().x, points.back().y + 1 };
+					direction = 1;
+					break;
+				}
+			}
+			// check pixel to left
+			if (points.back().x > 0 && direction != 3) {
+				if (river_data[idx - 1] == size + 1) {
+					points.push_back({ points.back().x - 1, points.back().y });
+					direction = 4;
+					continue;
+				} else if (river_data[idx - 1] == MERGE_COLOUR) {
+					merge_location = { points.back().x - 1, points.back().y };
+					merge = true;
+				} else if (river_data[idx - 1] > 1 && river_data[idx - 1] < 12) { // new segment
+					new_point = { points.back().x - 1, points.back().y };
+					direction = 4;
+					break;
+				}
+			}
+
+			// no further points
+			if (merge) points.push_back(merge_location);
+			river_complete = true;
+			break;
+		}
+
+		// save memory & simplify by storing only start, corner, and end points.
+		const auto is_corner_point = [](ivec2_t previous, ivec2_t current, ivec2_t next) {
+			return ((current.x - previous.x) * (next.y - current.y)) != ((current.y - previous.y) * (next.x - current.x)); // slope is fun
+		};
+		std::vector<ivec2_t> simplified_points;
+		simplified_points.push_back(points.front()); // add starting point
+		for (int i = 1; i < points.size()-1; ++i) {
+			if (is_corner_point(points[i-1], points[i], points[i+1])) { // add corner points
+				simplified_points.push_back(points[i]);
+			}
+		}
+		simplified_points.push_back(points.back());
+
+		// add segment then recursively call if neeeded
+		river.push_back({ size, std::move(simplified_points) });
+		if (river_complete) return;
+		next_segment(new_point, direction, river);
+	};
+
+	// find every river source and then run the segment algorithm.
+	for (int y = 0; y < rivers_bmp.get_height(); ++y) {
+		for (int x = 0; x < rivers_bmp.get_width(); ++x) {
+			if (river_data[x + y * rivers_bmp.get_width()] == START_COLOUR) { // start of a river
+				river_t river;
+
+				next_segment({ x, y }, 0, river);
```

Comment:
If I understand correctly the start point `{ x, y }` itself isn't included in the river, is it not needed for drawing the river or calculating crossings? If it is needed please include it in the river object

Reply:
V2 doesn't appear to include the start pixel, 
<img width="661" alt="image" src="https://github.com/user-attachments/assets/d5e52c58-9fa9-4d52-a66b-7fbb3e1805c7">
<img width="375" alt="image" src="https://github.com/user-attachments/assets/869b9ffd-f04f-4c0c-9cd4-91cad0ed162c">